simulator/src/main/java/com/hedera/block/simulator/grpc/impl/ConsumerStreamObserver.java

```diff
@@ -70,6 +74,9 @@ public ConsumerStreamObserver(
     @Override
     public void onNext(SubscribeStreamResponse subscribeStreamResponse) {
         final SubscribeStreamResponse.ResponseCase responseType = subscribeStreamResponse.getResponseCase();
+        if (lastKnownStatuses.size() == lastKnownStatusesCapacity) {
+            lastKnownStatuses.removeFirst();
```

Comment:
The javadoc for `removeFirst()` indicates that it will throw an exception if the Deque is empty. `pollFirst()` has the same behavior but it will just return `null` instead of throwing an exception. I'm not sure of the behavior we want for the simulator. What do you think?
<img width="325" alt="image" src="https://github.com/user-attachments/assets/bf97ec2e-9f45-4770-b944-bd7cdae3cd6c" />